packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -317,14 +317,36 @@ class AndroidSdk {
 
   String get avdManagerPath => getAvdManagerPath();
 
+  Directory get _platformsDir => fs.directory(fs.path.join(directory, 'platforms'));
+
+  Iterable<Directory> get _platforms {
+    Iterable<Directory> platforms = <Directory>[];
+    if (_platformsDir.existsSync()) {
+      platforms = _platformsDir
+        .listSync()
+        .whereType<Directory>();
+    }
+    return platforms;
+  }
+
   /// Validate the Android SDK. This returns an empty list if there are no
   /// issues; otherwise, it returns a list of issues found.
   List<String> validateSdkWellFormed() {
     if (!processManager.canRun(adbPath))
       return <String>['Android SDK file not found: $adbPath.'];
 
-    if (sdkVersions.isEmpty || latestVersion == null)
-      return <String>['Android SDK is missing command line tools; download from https://goo.gl/XxQghQ'];
+    if (sdkVersions.isEmpty || latestVersion == null) {
+      final StringBuffer msg = StringBuffer('No valid Android SDK platforms found in ${_platformsDir.path}.');
+      if (_platforms.isEmpty) {
+        msg.write('  Directory was empty. ');
```

Comment:
No need for trailing space here.  And you can go with one leading space to match the one below (or two in both - doesn't really matter as long as we're consistent).